From 48aecb81a4293b839ec8eb4bfe689ececfd0b300 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 21 May 2018 15:07:06 +0300
Subject: [PATCH 3829/5242] MLK-18357-2: drm/bridge: nwl-dsi: Add no_clk_reset
 property

commit  71ef9b97a3e2a2721890452c52db3482ce557967 from
https://source.codeaurora.org/external/imx/linux-imx.git

On some platforms, like the i.MX8M, the Display Controller is not
completely powered off during suspend, just stopped. Since there are
problems if we stop the clocks in DSI sub-system, while the Display
Controller is still powered on, the display clocks will get out of sync.
Adding this new property to specify, on such platforms, not to reset the
clocks.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/nwl-dsi.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/bridge/nwl-dsi.c b/drivers/gpu/drm/bridge/nwl-dsi.c
index 5ed93df..494379d 100644
--- a/drivers/gpu/drm/bridge/nwl-dsi.c
+++ b/drivers/gpu/drm/bridge/nwl-dsi.c
@@ -207,6 +207,7 @@ struct nwl_mipi_dsi {
 	u32				lanes;
 	u32				vc;
 	unsigned long			dsi_mode_flags;
+	bool				no_clk_reset;
 	bool				enabled;
 };
 
@@ -1080,6 +1081,9 @@ static void nwl_dsi_bridge_disable(struct drm_bridge *bridge)
 	phy_power_off(dsi->phy);
 	phy_exit(dsi->phy);
 
+	if (!dsi->no_clk_reset)
+		nwl_dsi_disable_clocks(dsi, CLK_PHY_REF | CLK_TX_ESC);
+
 	devm_free_irq(dev, dsi->irq, dsi);
 
 	dsi->enabled = false;
@@ -1162,6 +1166,8 @@ static int nwl_dsi_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 
+	dsi->no_clk_reset = of_property_read_bool(dev->of_node, "no_clk_reset");
+
 	dsi->dev = dev;
 	platform_set_drvdata(pdev, dsi);
 
-- 
1.7.9.5

